### PR TITLE
support protobuf build by default

### PR DESCRIPTION
**Pull request by BusyJay**
And support prost optionally

This is a derive work of #262 and #260.

### CODE REVIEW

**Comment by Hoverbear** [comment]
@BusyJay Seems there are some clippy errors. :(

**Comment by nrc** [comment]
Could you summarize how this different from #262 and #260 please? Why does this block #263 rather than #263 blocking this?

**Comment by BusyJay** [comment]
This PR uses protobuf-build to generate codes just as #262 does, but the generated codes are put inside target directory instead which is the same as #260.

**Comment by BusyJay** [comment]
Because CI in #260 is broken due to the changes to grpcio-compiler is out of sync to protobuf-build. This PR updates protobuf-build to a compatible version.

**Comment by hicqu** [comment]
@BusyJay are you fixing the CI?

**Comment by BusyJay** [comment]
Yes.

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +31,10 @@ install:
 script:
   - if [[ $TRAVIS_RUST_VERSION == "stable" && $TRAVIS_OS_NAME == "linux" ]]; then cargo fmt -- --check; fi
   - if [[ $TRAVIS_RUST_VERSION == "stable" && $TRAVIS_OS_NAME == "linux" ]]; then cargo clippy -- -D clippy::all; fi
+  - if [[ $TRAVIS_RUST_VERSION == "stable" && $TRAVIS_OS_NAME == "linux" ]]; then cargo clippy --no-default-features --features prost-codec -- -D clippy::all; fi
   - cargo test --all -- --nocapture
+  # There is a bug in protobuf-build that cached size is not supported yet.
```
Comment (overvenus) [inline .travis.yml]: Is the bug still here?

Comment (BusyJay) [reply .travis.yml]: Yes. It can be fixed later. Making rust-protobuf back is with highest priority.
